src/aws-bedrock/application-inference-profile.ts

```diff
@@ -0,0 +1,294 @@
+import { IResource, Resource, ResourceProps, Names } from 'aws-cdk-lib';
+import { CfnApplicationInferenceProfile } from 'aws-cdk-lib/aws-bedrock';
+import * as iam from 'aws-cdk-lib/aws-iam';
+import { Construct } from 'constructs';
+import { InferenceProfileModelSourceProps } from './model-source';
+
+type ConditionValue = string | number | boolean | string[] | number[];
+
+/**
+ * Options for granting permissions to invoke Bedrock foundation models via an inference profile
+ */
+export interface GrantInvokeOptions {
+  /**
+   * Whether to allow direct access to foundation models without requiring the use of this inference profile
+   * @default false - Models can only be accessed via this inference profile
+   */
+  readonly allowModelsDirectAccess?: boolean;
+
+  /**
+   * Model ARN pattern used to restrict access to foundation models.
+   * Useful when supporting multiple models.
+   * @default 'arn:aws:bedrock:*::foundation-model/*' (all foundation models)
+   */
+  readonly foundationModelArn?: string;
+
+  /**
+   * Additional conditions to match resource tags for tag-based access control.
+   * Keys are resource tag keys, values are principal tag variables or literal values.
+   * @example {'UserEmail': '${aws:PrincipalTag/UserEmail}'}
+   */
+  readonly tagConditions?: Record<string, string>;
+}
+
+/**
+ * Properties for defining a Bedrock Application Inference Profile
+ */
+export interface ApplicationInferenceProfileProps extends ResourceProps {
+  /**
+   * The name of the inference profile.
+   *
+   * @default - Assigned by CloudFormation (recommended).
```

Comment:
Actually it will be generated by CDK.

```suggestion
   * @default - Automatically generated.
```